V8 transport upgrade guide (#5208)
Particular/docs.particular.net: nservicebus/upgrades/7to8/index.md

NOTE: This is a working document; there is currently no timeline for the release of NServiceBus version 8.0.

## Transport configuration

NServiceBus V8 comes with a new transport configuration API. Instead of the generic based `UseTransport<TTransport>` method, create an instance of the transport's configuration class and pass it to `UseTransport`, e.g.

```csharp
var transport = endpointConfiguration.UseTransport<MyTransport>();

transport.Transactions(TransportTransactionMode.ReceiveOnly);
var routing = t.Routing();
routing.RouteToEndpoint(typeof(MyMessage), "DestinationEndpoint");
```

becomes:

```csharp
var transport = new MyTransport{
    TransportTransactionMode = TransportTransactionMode.ReceiveOnly
};

var routing = endpointConfiguration.UseTransport(transport);
routing.RouteToEndpoint(typeof(MyMessage), "DestinationEndpoint");
```

The transport can be configured via the transport's configuration type which derives from `TransportDefinition`. See the [transport upgrade guide](/nservicebus/upgrades/7to8/transport.md) for further details.

> ## Connection strings

Author: should this section be moved to the transport specific page or do we keep this here?

Reviewer: I think so, especially because there are transports for which a connection string doesn't make sense.

> Configuring a transport's connection using `.ConnectionStringName(name)`, which was removed for .NET Core in NServiceBus version 7, has been removed all platforms in NServiceBus version 8. To continue to retrieve the connection string by the named value in the configuration, first retrieve the connection string and then pass it to the `.ConnectionString(value)` configuration.

Reviewer: ```suggestion
Configuring a transport's connection using `.ConnectionStringName(name)`, which was removed for .NET Core in NServiceBus version 7, has been removed from all platforms in NServiceBus version 8. To continue to retrieve the connection string by the named value in the configuration, first retrieve the connection string and then pass it to the `.ConnectionString(value)` configuration.
```

Isn't `ConnectionString` gone?

* Azure Storage Queues
* SQL Transport
* SQS

## Outbox configuration

> NServiceBus version 8 requires explicit configuration of the transport transaction mode when using [Outbox](/nservicebus/outbox/):

Reviewer: ```suggestion
NServiceBus version 8 requires the transport transaction mode to be explicitly set to `ReceiveOnly` when using [Outbox](/nservicebus/outbox/):
```

@tmasternak thoughts 👆🏻 ?

```csharp
var transport = endpointConfiguration.UseTransport<MyTransport>();

> transport.TransportTransactionMode = TransportTransactionMode.ReceiveOnly;

Reviewer: Does it make sense to specify, if I understood correctly, that `ReceiveOnly` is the only supported mode when using the Outbox?
